docs/deployment/self-hosted/docker-containers.mdx

```diff
@@ -0,0 +1,142 @@
+```
+
+### Step 3. Build your supergraph
+
+```sh title="Using the deployment context, use the DDN CLI to create your project's metadata:"
+ddn supergraph build local
+```
+
+This will generate a set of JSON configuration files in `/engine/build` using your `.env.deployment` file.
+
+:::info Commit this to version control (optional)
+
+As you'll be using this metadata to build your Docker images on your server(s), you'll need a method by which you can
+copy these source files over. We recommend using version control (such as Git), but you can also SSH into your server
+and use `scp` to transfer over your project.
+
+:::
+
+## Build and run the engine
+
+### Step 1. Build your engine's image
+
+```bash title="On your server, with Docker installed and your latest metadata available, run the following command from the root of your project:"
+docker build -t my-engine -f engine/Dockerfile.engine engine
+```
+
+This will create an image named `my-engine` using the engine's Dockerfile.
+
+### Step 2. Run your engine
+
+```sh title="Then, from the root of your project, start the container with the required env vars:"
+docker run --rm -p 3280:3000 \
+  -e AUTHN_CONFIG_PATH=/md/auth_config.json \
+  -e ENABLE_CORS=true \
+  -e ENABLE_SQL_INTERFACE=true \
+  -e INTROSPECTION_METADATA_FILE=/md/metadata.json \
+  -e METADATA_PATH=/md/open_dd.json \
+  -e OTLP_ENDPOINT=<your_otlp_collector_endpoint> \
+  --add-host=local.hasura.dev:host-gateway \
+  my-engine
+```
+
+```sh title="You can verify this by sending a schema introspection request to your API:"
+curl -X POST https://<your-domain-or-ip>:3280/graphql \
+     -H "Content-Type: application/json" \
+     -d '{"query": "{ __schema { types { name } } }"}'
+```
+
+Of course, without a connector built and running, you won't be able to query data from the API.
+
+## Build and run a connector
+
+This process will need to be repeated for **each** connector.
+
+### Step 1. Build your connector's image
+
+```bash title="On your server, with Docker installed and the latest metadata available, run the following command from the root of your project:"
+docker build -t my-connector -f <subgraph_name>/connector/<connector_name>/.hasura-connector/Dockerfile.<connector_name> app/connector/<connector_name>/
+```
+
+This will create an image named `my-connector` using the connector's Dockerfile.
+
+### Step 2. Run your connector
+
+```sh title="Then, from the root of your project, start the container with the required env vars:"
+docker run --rm -p 9758:8080 \
+  -e CONNECTION_URI="<your_database_connection_uri>" \
+  -e HASURA_SERVICE_TOKEN_SECRET="<your_service_token_secret>" \
+  -e OTEL_EXPORTER_OTLP_ENDPOINT="<your_otlp_collector_endpoint>" \
+  -e OTEL_SERVICE_NAME="<subgraph_name>_<my_connector>" \
+  --add-host=local.hasura.dev:host-gateway \
+  my-connector
+```
+
+:::info Where do I find these values?
+
+You can find these values in the `.env.deployment` file you created earlier.
+
+:::
```

Comment:
Can we do a `### Step 3. Test your setup` or something like that to round it off? Something like the earlier curl command but with data. Or just something abstract like "Test your setup... should be working with multiple running containers". :)